src/file_pipe_log/reader.rs

```diff
@@ -64,17 +65,60 @@ impl<F: FileSystem> LogItemBatchFileReader<F> {
     /// Returns the next [`LogItemBatch`] in current opened file. Returns
     /// `None` if there is no more data or no opened file.
     pub fn next(&mut self) -> Result<Option<LogItemBatch>> {
-        if self.valid_offset < self.size {
+        // @lucasliang.
+        // TODO: [Fulfilled in writing progress when DIO is open.]
+        // We should also consider that there might exists broken blocks when DIO
+        // is open, and the following reading strategy should tolerate reading broken
+        // blocks until it finds an accessible header of `LogBatch`.
+        while self.valid_offset < self.size {
+            debug_assert!(self.file_context.is_some());
+            let data_layout = self.file_context.as_ref().unwrap().format.data_layout();
             if self.valid_offset < LOG_BATCH_HEADER_LEN {
                 return Err(Error::Corruption(
                     "attempt to read file with broken header".to_owned(),
                 ));
             }
-            let (footer_offset, compression_type, len) = LogBatch::decode_header(&mut self.peek(
-                self.valid_offset,
-                LOG_BATCH_HEADER_LEN,
-                0,
-            )?)?;
+            let (footer_offset, compression_type, len) = {
+                match data_layout {
```

Comment:
Use [if_chain](https://docs.rs/if_chain/latest/if_chain/) to reduce the nesting depth.
```rust
let r = LogBatch::decode_header(peek(valid_offset, HEADER_LEN, 0));
if_chain! {
    if r.is_err();
    if let DataLayout::Alignment(block_size) = data_layout;
    if block_size > 0;
    let aligned_next_offset = round_up(self.valid_offset, block_size);
    if valid_offset != aligned_next_offset;
    if is_padding(peek(valid_offset, aligned_next_offset - valid_offset, 0));
    then {
        continue;
    }
}
let (footer_offset, compression_type, len) = r?;
```